spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeTrainer.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin.spark;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.SparkExecutionPluginContext;
+import co.cask.cdap.etl.api.batch.SparkPluginContext;
+import co.cask.cdap.etl.api.batch.SparkSink;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.primitives.Doubles;
+import joptsimple.internal.Strings;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.linalg.Vectors;
+import org.apache.spark.mllib.regression.LabeledPoint;
+import org.apache.spark.mllib.tree.DecisionTree;
+import org.apache.spark.mllib.tree.model.DecisionTreeModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Spark Sink plugin that trains a model based upon a label in the structured record using Decision Tree Regression.
+ * Writes this model to a FileSet.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(DecisionTreeTrainer.PLUGIN_NAME)
+@Description("Trains a regression model based upon a particular label and featuresToInclude of a record.")
+public class DecisionTreeTrainer extends SparkSink<StructuredRecord> {
+  public static final String PLUGIN_NAME = "DecisionTreeTrainer";
+  //Impurity measure of the homogeneity of the labels at the node. Expected value for regression is "variance".
+  private static final String IMPURITY = "variance";
+  private static final int DEFAULT_MAX_DEPTH = 10;
+  private static final int DEFAULT_MAX_BINS = 100;
```

Comment:
its a little cleaner for these defaults to be in the config. In the no-args constructor for the config, you can set these values.